ecosystem/typescript/sdk/src/utils/memoize-decorator.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * Credits to https://github.com/darrylhodgins/typescript-memoize
+ */
+
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-restricted-syntax */
+
+interface MemoizeArgs {
```

Comment:
Can we add comments explaining what each of these fields do?

For example for `duration` (see suggestion below), is this in seconds? An absolute time? etc.

Reply:
duration renamed